Antinoblium will decay the station.

Pickle-Coding opened the pull request:
## About The Pull Request
Antinoblium will decay the station by rusting it during reaction. This is halted by hypernoblium as usual. It will not destroy walls.

## Why It's Good For The Game
Makes antinoblium look more dangerous while it spreads around the station. Gives a better impression that this gas decays everything.

## Changelog

:cl:
balance: Antinoblium decays the station.
/:cl:

Pickle-Coding committed:
Decay the station.

Antinoblium will decay the station by calling rust_turf() and rust_heretic_act() on the turfs and neighbouring closed turfs they are in during react().

Pickle-Coding commented on code/modules/atmospherics/gasmixtures/reactions.dm:
```diff
@@ -1261,5 +1261,19 @@
 		air.temperature = max(air.temperature * heat_capacity / new_heat_capacity, TCMB)
 
+	// Rusts its surroundings.
+	if(SPT_PROB(1, antinoblium_moles / 100) && isturf(holder)) // 1% chance per tick at 100 moles.
```
It's mathematically the same it's just the naming of the proc implies time based when it doesn't necessarily need to be.

LemonInTheDark replied:
yea s what I'm concerned about, masks intention I suppose? makes it harder to parse then literally just the math

SmArtKar commented on code/modules/atmospherics/gasmixtures/reactions.dm:
```diff
@@ -1261,5 +1261,19 @@
+			if(!HAS_TRAIT(turf, TRAIT_RUSTY)) // prevents destroying walls
+				turf.rust_turf()
+			for(var/obj/obj in turf)
+				obj.rust_heretic_act()
```
This will instakill most machinery and structures, as they just straight up take 500 damage from it. Did you even test this?

Pickle-Coding replied:
I decided to just not rust objects as I realised it will make fixing it almost impossible even if you contained it.

SmArtKar reviewed (changes requested):
This will turn antinoblium into an insane grief tool, rusting objects just destroys them 99% of the time, and rusting all walls will make the station look immensely ugly.
Also please fill out your WITGFTG with proper reasoning beyond "vibes"

Pickle-Coding committed:
Only rust turfs.

Only rust turfs by not rusting objects. Uses a single for loop instead of two.

LemonInTheDark commented:
what does this look like effect wise? can I see a video?

moocowswag commented:
I think this is cooler than rust ascension, it would be neat for rust heretics to be rewarded for knowing atmos.

Pretty much every heretic path really lacks any rewards for departmental knowledge outside of basic healing and bombing.

Pickle-Coding reviewed (commented):
> This will turn antinoblium into an insane grief tool, rusting objects just destroys them 99% of the time, and rusting all walls will make the station look immensely ugly.
> Also please fill out your WITGFTG with proper reasoning beyond "vibes" 

I updated my Why It's Good For The Game section.

Pickle-Coding reviewed (commented):
Incredible. What does that have to do with this PR though?

glue0000 commented:
> Incredible. What does that have to do with this PR though?

i think your PRs are not in good faith

Pickle-Coding reviewed (commented):
Fairly unsubstantial remark.

Pickle-Coding committed:
Math instead of SPT_PROB()

Replaces the usage of SPT_PROB() in /datum/gas_reaction/antinoblium_replication/react() with the same math.
